test/unit-test/FreeRTOS_ARP/FreeRTOS_ARP_utest.c

```diff
@@ -592,7 +584,7 @@ void test_eARPProcessPacket_Request_SenderMACSameAsLocalMAC( void )
 
     FreeRTOS_FindEndPointOnNetMask_IgnoreAndReturn( 1 );
     eResult = eARPProcessPacket( pxNetworkBuffer );
-    TEST_ASSERT_EQUAL( eReturnEthernetFrame, eResult );
+    TEST_ASSERT_EQUAL( eReleaseBuffer, eResult );
```

Comment:
This test was passing incorrectly, as [here](https://github.com/FreeRTOS/FreeRTOS-Plus-TCP/blob/6aa48a566e2259488179397b77c00837fa2eca07/source/FreeRTOS_ARP.c#L298) you should only get eReturnEthernetFrame if the MACs are different. And this test case was intending for them to be the same.